src/Symfony/Component/DependencyInjection/ContainerBuilder.php

```diff
@@ -1090,6 +1092,17 @@ public function resolveServices($value)
             foreach ($value as $k => $v) {
                 $value[$k] = $this->resolveServices($v);
             }
+        } elseif ($value instanceof LocatorArgument) {
+            $references = $value->getValues();
+            $value = new ServiceLocator(function ($id) use ($references) {
+                return isset($references[$id]) || in_array($id, array_map(function ($ref) { return (string) $ref; }, $references), true);
```

Comment:
the in_array part looks wrong to me - keys must be the ony way to reference values